src/js/components/FormField/FormField.js

```diff
@@ -156,27 +192,41 @@ class FormFieldContent extends Component {
             this.childContainerRef = ref;
           }}
           border={
-            border.position === 'inner'
-              ? {
+            Array.isArray(border)
+              ? this.createBorder(border).innerBorder
+              : border !== false &&
+                (border.position === 'inner' ||
+                  border.position === undefined) && {
                   ...border,
                   side: border.side || 'bottom',
                   color: borderColor,
                 }
-              : undefined
           }
         >
           {normalizedChildren}
         </Box>
       );
 
-      abut =
-        border.position === 'outer' &&
-        (border.side === 'all' || border.side === 'horizontal' || !border.side);
+      abut = Array.isArray(border)
+        ? this.checkAbut(border)
+        : border.position === 'outer' &&
+          (border.side === 'all' ||
+            border.side === 'horizontal' ||
+            !border.side);
       if (abut) {
         // marginBottom is set to overlap adjacent fields
         abutMargin = { bottom: '-1px' };
         if (margin) {
           abutMargin = margin;
+        } else if (Array.isArray(border)) {
```

Comment:
why does the border rendering is coupled to the `margin` assignment?

Reply:
This is existing from the previous code, but essentially this section is to help align FormFields when they are vertically stacked and use the size of the border to make them flush with each other. However, if the user specifies margin as a prop to FormField, it will use that value instead of the border size. Nothing new here about coupling with margin except that I made it able to handle when the border provided is an array.